docs/Porting_Guide.md

```diff
@@ -47,3 +48,116 @@ Understanding the TizenRT code structure and adhering to the structure helps in
+
+#### 6. Secure header
+This is the secure header signature. It is of type char and its size may vary depending upon the securing method.
+Note: The size of the secure header and the entire binary may vary depending on the secure method in the future.
+
+#### 7. Binary type
+This is the type of binary format say elf, bin etc. It is of size 1 byte.
+
+#### 8. Compression
+This is type of compression used on the binary i.e lzma, miniz or none. It is of size 1 byte.
+
+#### 9. Binary priority
+This is the priority of the main task in the binary. It is of size 1 byte.
+
+#### 10. Loading priority
+This is the priority of loading/reloading a binary i.e  HIGH, MID or LOW. It is of size 1 byte.
+
+#### 11. Binary name
+This is the name of the binary. It will correspond to the name of the main task in the binary. It is of size 16 bytes.
+
+#### 12. Binary version
+This is the version of binary. The binary manager checks the version and loads binary with the latest version. It is of size 4 bytes.
+
+#### 13. Binary ram size
+This is the size that will be needed by all the tasks in the binary. It is of size 4 bytes.
+
+#### 14. Binary stack size
+This is the stack size of the main task in the binary. It is of size 4 bytes.
+
+#### 15. Kernel Version
+This is the kernel version for compatibility with kernel. It is of size 2 bytes.
+
+## Partitions
+
+								   Binary 1			Binary 2
+									|			    |
+								 ----------------------------	 ---------------
+								| Header|    TizenRT Binary  |	| H |		|
+	 --------------------------------------------------------------------------------------------------------------------
+	|  km0 boot  |  km4 boot  |  reserved  | secure storage |	|	kernel		|	kernel		| .. |
+	 --------------------------------------------------------------------------------------------------------------------
+	|     16     |	    8     |     16     |       472      |	|	 1792		|	 1792		| .. |
+	 --------------------------------------------------------------------------------------------------------------------
+
+			Figure: Flash partition information of flash for Realtek board
+
+#### 1. Bootloader
+The bootloader section is placed at the start of the flash. In the above example realtek comprises of two bootloaders namely *km0 boot* & *km4 boot* of sizes 16 and 8 bytes respectively.
+
+#### 2. Reserved
+The reserved partition is next to bootloader. Its size is 16 bytes.
+
+#### 3. Secure Storage
+The secure storage partitions placed next to the reserved partition in the flash. Its 472 bytes in size.
+
+#### 4. Binary
+The header is placed at the start of the binary. Post the binary header lies the TizenRT kernel binary. In the above example realtek comprises of two kernel binaries of sizes 1792 bytes respectively.
+
+### To add binary header, user should take the following steps:
+
+1. User should define MAKE_SAMSUNG_HEADER for binary header support in each board in the following location:
```

Comment:
User should define MAKE_SAMSUNG_HEADER for binary header support in each board in the following location: build/configs/<board>/<name>/Make.defs
```
KERNEL_BINARY_PATH='xxxx.bin'
SECURE_HEADER_SIZE=xx

define MAKE_SAMSUNG_HEADER
  $(TOPDIR)/../os/tools/mksamsungheader.py ${KERNEL_BINARY_PATH} kernel ${SECURE_HEADER_SIZE}
endef
```
KERNEL_BINARY_PATH is blah blah
SECURE_HEADER_SIZE is blah blah.

Reply:
modified